src/pages/PassportRedemption/GiveawayPopover.tsx

```diff
@@ -0,0 +1,237 @@
+import React , { useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+import GreenCheck from './GreenCheck'
+import InstagramEnabled from './InstagramEnabled'
+import InstagramDisabled from './InstagramDisabled';
+
+import { getAllParticipatingSellers, getGiveawayTicketsForContact } from '../../utilities/api/interactionManager'
+
+
+interface Props {
+  showInstagram : boolean
+  contactId: string
+}
+
+interface TextProps {
+  icon ?: boolean
+}
+
+interface IgButtonProps {
+  disabled ?: boolean
+}
+interface GiveawayData {
+  weekly_giveaway_entries: String
+  number_of_visits_left : undefined | number
+}
+
+const GiveawayPopover = (props: Props) => {
+  const [showInfo, setShowInfo] = useState(false);
+  const [giveawayData, setData] = useState<GiveawayData>({
+    weekly_giveaway_entries: '',
+    number_of_visits_left: undefined
+  })
+  useEffect(() => {
+    async function setGiveawayCount(contactId: string) {
+      try {
```

Comment:
I've seen this in our existing passport pages but I still don't know why we do a try catch when the api call already catches an error

Reply:
Yeah, if the underlying InteractionManager function already handles the error, no need for this. That being said if you want to do something on error state, feel free to remove the error handling from the axios call--just make sure you leave a comment and make sure that it's in a try/catch everywhere it's called.